metaflow/plugins/kfp/kfp.py

```diff
@@ -404,19 +418,48 @@ def step_op_func(
     print("----")
 
     # TODO: Map username to KFP specific user/profile/namespace
+    # Note: we don't put this in a try catch block as below because
+    # Popen will not produce a error that will cause the Kubeflow step
+    # to stop
     with Popen(
         ["/bin/sh", "-c", cmd],
         stdin=PIPE,
         stdout=PIPE,
         stderr=STDOUT,
         universal_newlines=True,
         env=dict(os.environ, USERNAME="kfp-user", METAFLOW_RUN_ID=kfp_run_id),
-    ) as process, StringIO() as string_buffer:
+    ) as process, StringIO() as stdout_buffer:
         for line in process.stdout:
             print(line, end="")
-            string_buffer.write(line)
-        buffer_output = string_buffer.getvalue()
-    print("___ DONE ___")
+            stdout_buffer.write(line)
+        stdout_output = stdout_buffer.getvalue()
+
+    with open("0.stdout.log", "w") as stdout_file, open(
+        "0.stderr.log", "w"
+    ) as stderr_file:
+        _ = stdout_file.write(stdout_output)
+
+    # TODO: obtain the string `s3://kfp-example-aip-dev/metaflow/` dynamically
+    save_logs_cmd_template = (
+        f"python -m awscli s3 cp {{log_file}} {datastore_root}/"
+        f"{flow_name}/{kfp_run_id}/{step_name}/"
+        f"{task_id}/{{log_file}}"
+    )
+
+    log_stdout_cmd = save_logs_cmd_template.format(log_file="0.stdout.log")
+    log_stderr_cmd = save_logs_cmd_template.format(log_file="0.stderr.log")
```

Comment:
stderr is now removed?

Reply:
@talebzeghmi yup, now stdout contains both the standard output and standard error, which requires one less variable and one less buffer writing. The drawback is that all logs (including errors) are in stdout. Did you want something else?

Reply:
I would recommend not doing that, `stdout` is typically for end-user output whereas `stderr` is for any logging (incl. errors).

Reply:
Ok that will be changed back.